src/main/java/frc/robot/subsystems/AddressableLEDSubsystem.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import static frc.robot.Constants.LEDConstants.*;
+
+public class AddressableLEDSubsystem extends SubsystemBase {
+  /** Creates a new AddressableLEDSubsystem. */
+
+  AddressableLED LED;
+  AddressableLEDBuffer LEDBuffer;
+
+  public AddressableLEDSubsystem()
+  {
+    LED = new AddressableLED(PWM_PORT);
+    LEDBuffer = new AddressableLEDBuffer(LENGTH);
+    LED.setLength(LEDBuffer.getLength());
+    LED.setData(LEDBuffer);
+    LED.start();
+  }
+
+  public void LEDOff()
+  {
+  for(int i = 0; i < LEDBuffer.getLength(); i++)
+    {
+      LEDBuffer.setHSV(i, 0, 0, 0);
+    }
+  LED.setData(LEDBuffer);
+  }
+
+  public void HYC()
+  {
+    for(int i = 256; i < LEDBuffer.getLength(); i++)
+    {
+      LEDBuffer.setHSV(i, YELLOW_H, YELLOW_S, YELLOW_V);
+    }
+    LED.setData(LEDBuffer);
+  }
+
+  public void DYC()
+  {
+    for(int i = 0; i < 256; i++)
+    {
+      LEDBuffer.setHSV(i, YELLOW_H, YELLOW_S, YELLOW_V);
+    }
+    LED.setData(LEDBuffer);
+  }
+
+  public void HPC()
+  {
+    for(int i = 256; i < LEDBuffer.getLength(); i++)
+    {
+      LEDBuffer.setHSV(i, PURPLE_H, PURPLE_S, PURPLE_V);
+    }
+    LED.setData(LEDBuffer);
+  }
+
+  public void DPC()
+  {
+    for(int i = 0; i < 256; i++)
+    {
+      LEDBuffer.setHSV(i, PURPLE_H, PURPLE_S, PURPLE_V);
+    }
+    LED.setData(LEDBuffer);
+  }
+
+  public CommandBase HumanYellowColor()
+  {
+    return runEnd(
+    () ->
+    {
+      HYC();
+    },
+
+    () ->
+    {
+      LEDOff();
+    }
+    );
+  }
```

Comment:
lambdas with only one statement can be replaced with a one-line lambda or a method reference. i personally prefer method references but either is usable

one-line lambda example:
```java
() -> ledMethodHere();
```

method reference example:
```java
this::ledMethodHere
```